src/calibration_functions.jl

```diff
@@ -0,0 +1,331 @@
+@doc raw""" 
+    empirical_frequency_regression(Y_cal, distributions::Distributions.Normal, n_bins=20)
+Dispatched version for Normal distributions
+FOR REGRESSION MODELS.  \
+Given a calibration dataset ``(x_t, y_t)`` for ``i ∈ {1,...,T}`` and an array of predicted distributions, the function calculates the empirical frequency
+```math
+p^hat_j = {y_t|F_t(y_t)<= p_j, t= 1,....,T}/T,
+```
+where ``T`` is the number of calibration points, ``p_j`` is the confidence level and ``F_t`` is the 
+cumulative distribution function of the predicted distribution targeting ``y_t``. \
+Source: [Kuleshov, Fenner, Ermon 2018](https://arxiv.org/abs/1807.00263)
+
+Inputs: \
+    - `Y_cal`: a vector of values ``y_t``\
+    - `distributions`:a Vector{Distributions.Normal{Float64}} of distributions stacked row-wise.\
+        For example the output of LaplaceRedux.predict(la,X_cal)
+    - `n_bins`: number of equally spaced bins to use.\
+Outputs:\
+    - `counts`: an array cointaining the empirical frequencies for each quantile interval.
+"""
+function empirical_frequency_regression(
+    Y_cal, distributions::Vector{Normal{Float64}}; n_bins::Int=20
+)
+    if n_bins <= 0
+        throw(ArgumentError("n_bins must be a positive integer"))
+    end
+    n_edges = n_bins + 1
+    quantiles = collect(range(0; stop=1, length=n_edges))
+    quantiles_matrix = hcat(
+        [map(Base.Fix1(quantile, distr), quantiles) for distr in distributions]...
+    )#warning deprecated, need to change in not sure what
+    n_rows = size(quantiles_matrix, 1)
+    counts = Float64[]
+
+    for i in 1:n_rows
+        push!(counts, sum(Y_cal .<= quantiles_matrix[i, :]) / length(Y_cal))
+    end
+    return counts
+end
+
+@doc raw""" 
+    sharpness_classification(y_binary,distributions::Distributions.Bernoulli)
+dispatched for Bernoulli Distributions
+FOR BINARY CLASSIFICATION MODELS.  \
+Assess  the sharpness of the model by looking at the distribution of model predictions.  
+When forecasts are sharp, most predictions are close to either 0 or 1   \
+Source: [Kuleshov, Fenner, Ermon 2018](https://arxiv.org/abs/1807.00263)
+
+Inputs:  \
+    - `y_binary`: the array of outputs  ``y_t``  numerically coded: 1 for the target class, 0 for the negative result.  \
+    - `distributions`: an array of Bernoulli distributions describing the probability of of the output belonging to the target class \
+
+Outputs:  \
+    -  `mean_class_one`: a scalar that measure the average prediction for the target class  \
+    -  `mean_class_zero`: a scalar that measure the average prediction for the null class  
+
+"""
+function sharpness_classification(y_binary, distributions::Vector{Bernoulli{Float64}})
+    mean_class_one = mean(mean.(distributions[findall(y_binary .== 1)]))
+    mean_class_zero = mean(1 .- mean.(distributions[findall(y_binary .== 0)]))
+    return mean_class_one, mean_class_zero
+end
+
+@doc raw""" 
+    empirical_frequency_classification(y_binary, distributions::Distributions.Bernoulli)
```

Comment:
Are these docstrings rendering correctly in the REPL when you use help mode?

Reply:
i will check one for one. at the moment i am working on the multi.qmd file. i am failing to make to join the plots in a 2,2 grid. I will upload as it is now , so that you can see.

Reply:
btw i was considering deleting the original functions for sampled distributions and leaving only those for normal distributions since laplaceredux only works for this kind of distr. i will submit the general ones to the guys at calibration.jl @pat-alt